### PR TITLE
Updated crate_universe dependencies

### PR DESCRIPTION
**Pull request by UebelAndre**

### CODE REVIEW

File: crate_universe/src/rendering/templates/partials/crate/aliases.j2 [hunk 1]
```diff
@@ -1,9 +1,9 @@
 selects.with_or({
     {%- for cfg, values in selectable.selects %}
-    {%- if cfg in context.conditions and context.conditions[cfg] | length %}
+    {%- if context.conditions | get(key=cfg, default="") | length%}
```
Comment (UebelAndre) [inline crate_universe/src/rendering/templates/partials/crate/aliases.j2:3]: The changes here are to account for the change in behavior from https://github.com/Keats/tera/issues/738

File: crate_universe/src/metadata.rs [hunk 2]
```diff
@@ -76,7 +76,7 @@ impl MetadataGenerator for Generator {
 }
 
 /// A configuration desrcibing how to invoke [cargo update](https://doc.rust-lang.org/cargo/commands/cargo-update.html).
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
```
Comment (UebelAndre) [inline crate_universe/src/metadata.rs:79]: This seems related to https://github.com/clap-rs/clap/issues/4286 but is not a big deal for our uses here.

File: crate_universe/src/context/platforms.rs [hunk 3]
```diff
@@ -165,39 +165,58 @@ mod test {
...
             },
             ..CrateContext::default()
-        };
+        }
+    }
 
-        let configurations =
-            resolve_cfg_platforms(vec![&context], &supported_platform_triples()).unwrap();
+    #[test]
+    fn resolve_targeted() {
```
Comment (UebelAndre) [inline crate_universe/src/context/platforms.rs:191]: This test was refactored to gain higher confidence in it's validity. The real issue I was seeing was addressed by https://github.com/bazelbuild/rules_rust/pull/1591#discussion_r992959641

File: crate_universe/src/context/crate_context.rs [hunk 4]
```diff
@@ -284,7 +284,7 @@ impl CrateContext {
             crate_features: annotation.node.features.iter().cloned().collect(),
             deps,
             deps_dev,
-            edition: package.edition.clone(),
+            edition: package.edition.as_str().to_string(),
```
Comment (UebelAndre) [inline crate_universe/src/context/crate_context.rs:287]: This accounts for the thread of changes started in https://github.com/oli-obk/cargo_metadata/pull/183

File: crate_universe/src/config.rs [hunk 5]
```diff
@@ -481,17 +481,17 @@ mod test {
         let mut package = mock_cargo_metadata_package();
         let id = CrateId::new("mock-pkg".to_owned(), "0.1.0".to_owned());
 
-        package.version = cargo_metadata::Version::new(0, 1, 0);
+        package.version = cargo_metadata::semver::Version::new(0, 1, 0);
```
Comment (UebelAndre) [inline crate_universe/src/config.rs:484]: These changes account for https://github.com/oli-obk/cargo_metadata/issues/179

**Comment by UebelAndre** [comment]
There appears to have been a regression in clap v4. A discussion was spawned here:
- https://github.com/clap-rs/clap/discussions/4374

**Review by illicitonion — APPROVED** [review]

File: crate_universe/Cargo.toml [hunk 6]
```diff
@@ -20,26 +20,27 @@ default = ["cargo"]
 
 [dependencies]
 anyhow = "1.0.57"
-cargo_metadata = "0.14.2"
-cargo_toml = "0.11.5"
+# TODO: https://github.com/oli-obk/cargo_metadata/issues/208
```
Comment (UebelAndre) [inline crate_universe/Cargo.toml:23]: This obviously links to https://github.com/oli-obk/cargo_metadata/issues/208

**Pull request merged**